Nomad: FIPS-140-3 compliant binaries (#3002)
hashicorp/web-unified-docs: content/nomad/v1.10.x/content/docs/enterprise/fips.mdx

general guidance about using Nomad Enterprise in a FIPS-compliant manner. We
recommend consulting an approved auditor for further information.

The FIPS 140-3 variant of Nomad uses separate binaries that are available from
the following sources:

- The [HashiCorp Releases page](https://releases.hashicorp.com/nomad), releases
  ending with the `+ent.fips1403` suffix. Only Linux binaries are available.

### Usage restrictions

When using Nomad Enterprise with FIPS 140-3, be aware of the following operation
restrictions.

#### Migration restrictions

We do not support in-place migrations from non-FIPS builds of Nomad to FIPS
builds of Nomad, regardless of version. A fresh cluster installation is required
to receive support.

We generally recommend avoiding direct upgrades and replicated-migrations
because existing wrapped key material is encrypted with non-FIPS compliant
cryptographic modules, and all Nomad Variables and Workload Identities have been
encrypted or signed with this non-compliant key material. Additionally, service
IDs and other identifiers have been derived from non-compliant hash functions,
which may lead to workload service interruptions during migration.

> As such Hashicorp cannot provide support for workloads that are affected either

Reviewer: ```suggestion
As such HashiCorp cannot provide support for workloads that are affected either
```

technically or via non-compliance that results from converting existing cluster
workloads to the FIPS-compliant binary.

Instead, we suggest leaving the existing cluster in place, and carefully
consider migration of specific workloads to the FIPS cluster.

#### TLS restrictions

Nomad Enterprise's FIPS modifications include restrictions to supported TLS
cipher suites and key information. Only the following cipher suites are allowed
in TLS1.2 and below:

- `TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256`
- `TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384`
- `TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256`
- `TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384`
- `TLS_RSA_WITH_AES_128_GCM_SHA256`
- `TLS_RSA_WITH_AES_256_GCM_SHA384`

In addition, only the following key types are allowed in TLS chains of trust:

- RSA 2048, 3072, 4096, 7680, and 8192-bit
- ECDSA P-256, P-384, and P-521

#### Feature restrictions

The following features are not available or work in a modified way on FIPS mode

> binaries.

Reviewer: ```suggestion
binaries:
```

- The `nomad alloc exec` and `nomad job action` commands, and their related
  APIs, are not available and return an error message.
- OIDC configuration does not allow the `x5t` key ID header. Only the `x5t#S256`
  header is allowed.
- The `md5` and `sha1` HCL2 functions return an error if used in a job
  specification.
- The `md5` and `sha1` checksum functions for `artifact` blocks return an error
  if used in a job specification.

> - The `md5sum` function will return an error if used in the content of a

Reviewer: ```suggestion
- The `md5sum` function returns an error if used in the content of a
```
